tests/test_novoCaller.py

```diff
@@ -14,6 +15,66 @@
 # this tests may fail because of differences in float handling
 # e.g. novoCaller=1.4030838153054451e-05 Vs novoCaller=1.403083815305445e-05
 # depending on the machine
+
+def assert_(tfile, ofile, test=False):
+    ''' '''
+    # Creating Vcf object
+    vcf_obj_t = vcf_parser.Vcf(tfile)
+    vcf_obj_o = vcf_parser.Vcf(ofile)
+    # Check headers
+    assert vcf_obj_t.header.definitions == vcf_obj_o.header.definitions
+    # Get variants
+    vnt_obj_t_list, vnt_obj_o_list = [], []
+    for vnt_obj in vcf_obj_t.parse_variants(): vnt_obj_t_list.append(vnt_obj)
+    for vnt_obj in vcf_obj_o.parse_variants(): vnt_obj_o_list.append(vnt_obj)
+    # Check variants number is consistent
+    assert len(vnt_obj_t_list) == len(vnt_obj_o_list)
+    # Check variants
+    for i, vnt_obj_t in enumerate(vnt_obj_t_list):
+        vnt_obj_o = vnt_obj_o_list[i]
+        # Check basic fields are matching
+        assert vnt_obj_t.CHROM == vnt_obj_o.CHROM
+        assert vnt_obj_t.POS == vnt_obj_o.POS
+        assert vnt_obj_t.ID == vnt_obj_o.ID
+        assert vnt_obj_t.REF == vnt_obj_o.REF
+        assert vnt_obj_t.ALT == vnt_obj_o.ALT
+        assert vnt_obj_t.QUAL == vnt_obj_o.QUAL
+        assert vnt_obj_t.FILTER == vnt_obj_o.FILTER
+        assert vnt_obj_t.INFO.split(';')[:-2] == vnt_obj_o.INFO.split(';')[:-2]
+        assert vnt_obj_t.FORMAT == vnt_obj_o.FORMAT
+        # novoAF
+        #   since there may be minor differences in rounding
+        #   we check that the results are close enough to the expected
+        try:
+            af_t = float(vnt_obj_t.get_tag_value('novoAF'))
+            af_o = float(vnt_obj_o.get_tag_value('novoAF'))
+        except ValueError:
+            af_t = float(vnt_obj_t.get_tag_value('AlleleFreq'))
+            af_o = float(vnt_obj_o.get_tag_value('AlleleFreq'))
+        assert abs(af_t - af_o) < 0.0000000001
```

Comment:
Sometime we should talk about good techniques for floating point comparison like this, but you are using this technique more than once, so the minimum thing I'd do is factor it out into a separate function like (untested):
```
DEFAULT_APPROXIMATION_THRESHOLD = 0.0000000001
def approximately_equal(x, y, threshold=DEFAULT_APPROXIMATION_THRESHOLD):
    return abs(x - y) < threshold
```
so that if you find a better way, you don't have a ton of edits to do. I do think there are better ways to do this approximation test, but I'll have to go look it up. What's missing is a sense of the scale of numbers that are being compared. You need to compare very small floats differently than large ones, and there's a way to scale them to a normalized size and then apply the threshold to that which is recognized as better.